lib/handleRequest.js

```diff
@@ -14,14 +14,11 @@ function handleRequest (req, res, params, context) {
   var headers = req.headers
   var request = new context.Request(params, req, urlUtil.parse(req.url, true).query, headers, req.log)
   var reply = new context.Reply(res, context, request)
-
-  if (method === 'GET' || method === 'HEAD') {
-    return handler(reply)
-  }
-
   var contentType = headers['content-type']
 
-  if (method === 'POST' || method === 'PUT' || method === 'PATCH') {
+  if (method === 'GET' || method === 'HEAD') {
```

Comment:
I would keep this first `if` statement where it is, otherwise the `content-type` header is read when it doesn't need to be.